scico/functional/_denoiser.py

```diff
@@ -13,6 +13,34 @@
 from ._functional import Functional
 
 
+class BM4D(Functional):
+    r"""Pseudo-functional whose prox applies the BM4D denoising algorithm.
+
+    A pseudo-functional that has the BM4D algorithm
+    :cite:`maggioni2012nonlocal` as its proximal operator, which calls
+    :func:`.denoiser.bm4d`.
+    """
+
+    has_eval = False
+    has_prox = True
+
+    def __init__(self):
+        r"""Initialize a :class:`BM4D` object."""
+        super().__init__()
+
+    def prox(self, x: JaxArray, lam: float = 1.0, **kwargs) -> JaxArray:
+        r"""Apply BM4D denoiser.
+
+        Args:
+            x: Input image.
+            lam: Noise parameter.
```

Comment:
kwargs missing

Reply:
As in description of kwargs in the docstring? Neither bm3d nor dncnn has it though.

Reply:
I see your point: completely missing in all of the `denoiser` module. I think it should look like the `prox` docs in e.g. `NonNegativeIndicator`.

Reply:
Fixed it.